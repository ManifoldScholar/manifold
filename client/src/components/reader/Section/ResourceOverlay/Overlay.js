import React, { PureComponent, PropTypes } from 'react';
import ResourceDetail from './ResourceDetail';

export default class ResourceOverlay extends PureComponent {

  static propTypes = {
    resource: PropTypes.object
  };

  render() {
    return (
      <div className="overlay-full-secondary bg-neutral90">
        <div className="overlay-close">
          Close
          <i className="manicon manicon-x"></i>
        </div>
        <ResourceDetail
            resource={this.props.resource}
        />
      </div>
    );
  }
}
