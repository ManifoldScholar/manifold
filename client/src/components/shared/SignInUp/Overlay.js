import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';
import PasswordForgot from './PasswordForgot';
import PasswordReset from './PasswordReset';
import Login from './Login';
import Update from './Update';
import Create from './Create';
import { browserHistory } from 'react-router';

export default class Overlay extends Component {

  static propTypes = {
    visible: PropTypes.bool,
    hideSignInUpOverlay: PropTypes.func,
    authentication: PropTypes.object,
    dispatch: PropTypes.func,
    hash: PropTypes.string
  };

  constructor() {
    super();
    this.updateHash = this.updateHash.bind(this);
    this.childProps = this.childProps.bind(this);
  }

  componentDidUpdate(prevProps) {
    if (this.props.authentication.authenticated === true &&
      prevProps.authentication.authenticated === false) {
      this.props.hideSignInUpOverlay();
    }
  }

  updateHash(hash) {
    return () => {
      browserHistory.push({ ...location, hash: `#${hash}` });
    };
  }

  childProps() {
    return {
      updateHash: this.updateHash,
      dispatch: this.props.dispatch,
      authentication: this.props.authentication
    };
  }

  renderChild() {
    let child = null;
    const childProps = this.childProps();
    switch (this.props.hash) {
      case '#account-create':
        child = <Create {...childProps} />;
        break;
      case '#account-update':
        child = <Update {...childProps} />;
        break;
      case '#account-password-forgot':
        child = <PasswordForgot {...childProps} />;
        break;
      case '#account-password-reset':
        child = <PasswordReset {...childProps} />;
        break;
      case '#account-login':
      default:
        child = <Login {...childProps} />;
        break;
    }
    return child;
  }

  render() {
    const overlayClass = classNames({
      'overlay-login': true,
      'overlay-hidden': !this.props.visible,
      'overlay-visible': this.props.visible
    });
    return (
      <div className={overlayClass}>
        <figure className="logo">
          <i className="manicon manicon-manifold-logo"></i>
          Manifold
        </figure>
        <button onClick={this.props.hideSignInUpOverlay} className="overlay-close">
          Cancel Log in
          <i className="manicon manicon-x"></i>
        </button>
        <div className="overlay-content">
          {this.renderChild()}
        </div>
      </div>
    );
  }
}
