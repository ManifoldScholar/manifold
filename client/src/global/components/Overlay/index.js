import React, { useEffect, useRef } from "react";
import PropTypes from "prop-types";
import FocusTrap from "focus-trap-react";
import { useUID } from "react-uid";
import Header from "./Header";

import BodyClass from "hoc/BodyClass";

function Overlay({
  title,
  subtitle,
  icon,
  appearance,
  contentWidth,
  closeCallback,
  closeUrl,
  history,
  triggerScrollToTop,
  id,
  children,
  ariaLabel
}) {
  const overlayRef = useRef(null);
  const headerId = useUID();

  function handleCloseEvent(event) {
    if (closeCallback) {
      closeCallback(event);
    }
    if (closeUrl) {
      setTimeout(() => {
        history.push(closeUrl);
      }, 200);
    }
  }

  useEffect(() => {
    if (!overlayRef.current) return;
    overlayRef.current.scrollTop = 0;
  }, [triggerScrollToTop]);

  return (
    <BodyClass className="no-scroll overlay">
      <div
        className={appearance || "overlay-full"}
        key="globalOverlay"
        ref={overlayRef}
        id={id}
        role="dialog"
        aria-modal
        aria-labelledby={headerId}
      >
        <FocusTrap
          focusTrapOptions={{
            escapeDeactivates: e => handleCloseEvent(e)
          }}
        >
          <div>
            <Header
              title={title}
              subtitle={subtitle}
              ariaLabel={ariaLabel}
              icon={icon}
              onCloseClick={handleCloseEvent}
              headerId={headerId}
            />
            <div style={{ maxWidth: contentWidth }} className="container">
              {children}
            </div>
          </div>
        </FocusTrap>
      </div>
    </BodyClass>
  );
}

Overlay.displayName = "Global.Overlay";

Overlay.propTypes = {
  open: PropTypes.bool,
  history: PropTypes.object,
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
  title: PropTypes.string,
  subtitle: PropTypes.string,
  icon: PropTypes.string,
  closeUrl: PropTypes.string,
  closeCallback: PropTypes.func,
  contentWidth: PropTypes.number,
  appearance: PropTypes.string,
  triggerScrollToTop: PropTypes.any,
  id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  ariaLabel: PropTypes.string
};

export default Overlay;
