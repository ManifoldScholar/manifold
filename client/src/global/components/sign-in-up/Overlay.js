import React, { Component } from "react";
import PropTypes from "prop-types";
import FocusTrap from "focus-trap-react";
import Utility from "global/components/utility";
import BodyClass from "hoc/body-class";
import PasswordForgot from "./PasswordForgot";
import Login from "./Login";
import CreateUpdate from "./CreateUpdate";
import Update from "./Update";
import Create from "./Create";

export default class Overlay extends Component {
  static propTypes = {
    hideSignInUpOverlay: PropTypes.func,
    authentication: PropTypes.object,
    settings: PropTypes.object,
    dispatch: PropTypes.func
  };

  constructor(props) {
    super(props);
    this.state = {
      view: props.authentication.authenticated
        ? "account-update"
        : "account-login"
    };
  }

  componentDidUpdate(prevProps) {
    if (
      this.props.authentication.authenticated &&
      !prevProps.authentication.authenticated
    ) {
      if (this.state.view !== "account-create-update") {
        this.props.hideSignInUpOverlay();
      }
    }
  }

  updateView = (view, event = null) => {
    if (event) event.preventDefault();
    this.setState(Object.assign(this.state, {}, { view }));
  };

  childProps = () => {
    return {
      handleViewChange: this.updateView,
      settings: this.props.settings,
      dispatch: this.props.dispatch,
      hideSignInUpOverlay: this.props.hideSignInUpOverlay,
      authentication: this.props.authentication
    };
  };

  renderChild() {
    let child = null;
    const childProps = this.childProps();
    switch (this.state.view) {
      case "account-create":
        child = <Create {...childProps} />;
        break;
      case "account-update":
        child = <Update {...childProps} />;
        break;
      case "account-create-update":
        child = <CreateUpdate {...childProps} />;
        break;
      case "account-password-forgot":
        child = <PasswordForgot {...childProps} />;
        break;
      case "account-login":
        child = <Login {...childProps} />;
        break;
      default:
        child = null;
        break;
    }
    return child;
  }

  render() {
    return (
      <BodyClass className="no-scroll">
        <div className="overlay-full-primary">
          <FocusTrap
            focusTrapOptions={{
              onDeactivate: this.props.hideSignInUpOverlay
            }}
          >
            <header className="overlay-header">
              <div className="container">
                <div className="rel">
                  <figure className="header-logo" aria-hidden="true">
                    <i className="manicon" aria-hidden="true">
                      <Utility.IconComposer size={26} icon="manifoldLogo32" />
                    </i>
                  </figure>
                  <button
                    onClick={this.props.hideSignInUpOverlay}
                    className="overlay-close"
                    data-id="overlay-close"
                  >
                    Cancel
                    <i className="manicon manicon-x" aria-hidden="true" />
                  </button>
                </div>
              </div>
            </header>
            <div className="overlay-content focus">
              <div className="container">
                <div className="inner">{this.renderChild()}</div>
              </div>
            </div>
          </FocusTrap>
        </div>
      </BodyClass>
    );
  }
}
