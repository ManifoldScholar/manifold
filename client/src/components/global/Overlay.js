import React, { Component } from "react";
import PropTypes from "prop-types";
import { HigherOrder } from "components/global";
import FocusTrap from "focus-trap-react";
import isString from "lodash/isString";

export default class Overlay extends Component {
  static propTypes = {
    open: PropTypes.bool,
    history: PropTypes.object,
    children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
    title: PropTypes.string,
    icon: PropTypes.string,
    closeUrl: PropTypes.string,
    closeCallback: PropTypes.func,
    contentWidth: PropTypes.number,
    appearance: PropTypes.string,
    triggerScrollToTop: PropTypes.any
  };

  constructor() {
    super();
    this.state = {
      view: null
    };
  }

  componentDidUpdate(prevProps) {
    if (prevProps.triggerScrollToTop !== this.props.triggerScrollToTop) {
      this.scrollableEl.scrollTo(0, 0);
    }
  }

  handleCloseEvent = event => {
    if (this.props.closeCallback) {
      this.props.closeCallback(event);
    }
    if (this.props.closeUrl) {
      setTimeout(() => {
        this.props.history.push(this.props.closeUrl);
      }, 200);
    }
  };

  overlayClass() {
    return this.props.appearance ? this.props.appearance : "overlay-full";
  }

  renderChildren() {
    if (!this.props.children) return null;
    if (isString(this.props.children.type)) return this.props.children;
    return React.cloneElement(this.props.children, {
      closeDrawer: this.handleLeaveEvent
    });
  }

  renderTitle(icon, title) {
    return (
      <div className="container">
        <h3 className="overlay-title">
          <i className={`manicon manicon-${icon}`} aria-hidden="true" />
          {title}
        </h3>
      </div>
    );
  }

  renderHeader(props) {
    if (!props.title) return null;
    return (
      <header className="overlay-full-header" key={"globalOverlayHeader"}>
        {this.renderTitle(this.props.icon, this.props.title)}
        <button
          onClick={this.handleCloseEvent}
          className="overlay-close"
          data-id="overlay-close"
        >
          Close
          <i className="manicon manicon-x" aria-hidden="true" />
        </button>
      </header>
    );
  }

  render() {
    return (
      <HigherOrder.BodyClass className={"no-scroll overlay"}>
        <div
          className={this.overlayClass()}
          key={"globalOverlay"}
          ref={el => {
            this.scrollableEl = el;
          }}
        >
          <FocusTrap
            focusTrapOptions={{
              escapeDeactivates: false
            }}
          >
            {this.renderHeader(this.props)}
            {!this.props.title ? (
              <button
                onClick={this.handleCloseEvent}
                className="overlay-close"
                data-id="overlay-close"
              >
                Close
                <i className="manicon manicon-x" aria-hidden="true" />
              </button>
            ) : null}
            <div
              style={{ maxWidth: this.props.contentWidth }}
              className="container"
            >
              {this.renderChildren()}
            </div>
          </FocusTrap>
        </div>
      </HigherOrder.BodyClass>
    );
  }
}
