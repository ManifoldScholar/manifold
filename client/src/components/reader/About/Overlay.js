import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { About } from "components/reader";
import lh from "helpers/linkHandler";

export default class AboutOverlay extends PureComponent {
  static propTypes = {
    text: PropTypes.object,
    match: PropTypes.object,
    history: PropTypes.object
  };

  constructor(props) {
    super(props);
    this.handleClose = this.handleClose.bind(this);
  }

  handleClose(eventIgnored) {
    const { textId, sectionId } = this.props.match.params;
    const closeUrl = lh.link("readerSection", textId, sectionId);
    return this.props.history.push(closeUrl);
  }

  render() {
    return (
      <div className="overlay-full-secondary bg-neutral90">
        <div onClick={this.handleClose} className="overlay-close">
          Close
          <i className="manicon manicon-x" />
        </div>
        <About.Detail text={this.props.text} handleClose={this.handleClose} />
      </div>
    );
  }
}
