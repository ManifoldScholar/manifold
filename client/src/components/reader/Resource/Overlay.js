import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { Resource } from "components/reader";
import lh from "helpers/linkHandler";

export default class ResourceOverlay extends PureComponent {
  static propTypes = {
    resource: PropTypes.object,
    match: PropTypes.object,
    history: PropTypes.object
  };

  constructor(props) {
    super(props);
    this.handleClose = this.handleClose.bind(this);
  }

  handleClose(eventIgnored) {
    const { textId, sectionId } = this.props.match.params;
    const closeUrl = lh.link("readerSection", textId, sectionId);
    this.props.history.push(closeUrl);
  }

  render() {
    const { resource } = this.props;
    const resourceUrl = lh.link(
      "frontendProjectResource",
      resource.attributes.projectId,
      resource.id
    );

    return (
      <div className="overlay-full-secondary bg-neutral90">
        <div onClick={this.handleClose} className="overlay-close">
          Close
          <i className="manicon manicon-x" />
        </div>
        <Resource.Detail
          resource={resource}
          resourceUrl={resourceUrl}
          handleClose={this.handleClose}
        />
      </div>
    );
  }
}
