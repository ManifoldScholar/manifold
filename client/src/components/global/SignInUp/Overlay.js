import React, { Component } from "react";
import PropTypes from "prop-types";
import { HigherOrder } from "components/global";
import PasswordForgot from "./PasswordForgot";
import Login from "./Login";
import CreateUpdate from "./CreateUpdate";
import Update from "./Update";
import Create from "./Create";

export default class Overlay extends Component {
  static propTypes = {
    hideSignInUpOverlay: PropTypes.func,
    authentication: PropTypes.object,
    settings: PropTypes.object,
    dispatch: PropTypes.func
  };

  constructor(props) {
    super(props);
    this.state = {
      view: props.authentication.authenticated
        ? "account-update"
        : "account-login"
    };
    this.updateView = this.updateView.bind(this);
    this.childProps = this.childProps.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (
      this.props.authentication.authenticated === false &&
      nextProps.authentication.authenticated === true
    ) {
      if (this.state.view !== "account-create-update") {
        this.props.hideSignInUpOverlay();
      }
    }
  }

  updateView(view, event = null) {
    if (event) event.preventDefault();
    this.setState(Object.assign(this.state, {}, { view }));
  }

  childProps() {
    return {
      handleViewChange: this.updateView,
      settings: this.props.settings,
      dispatch: this.props.dispatch,
      hideSignInUpOverlay: this.props.hideSignInUpOverlay,
      authentication: this.props.authentication
    };
  }

  renderChild() {
    let child = null;
    const childProps = this.childProps();
    switch (this.state.view) {
      case "account-create":
        child = <Create {...childProps} />;
        break;
      case "account-update":
        child = <Update {...childProps} />;
        break;
      case "account-create-update":
        child = <CreateUpdate {...childProps} />;
        break;
      case "account-password-forgot":
        child = <PasswordForgot {...childProps} />;
        break;
      case "account-login":
        child = <Login {...childProps} />;
        break;
      default:
        child = null;
        break;
    }
    return child;
  }

  render() {
    return (
      <HigherOrder.BodyClass className={"no-scroll"}>
        <div className="overlay-full-primary">
          <header className="overlay-header">
            <div className="container">
              <div className="rel">
                <figure className="logo">
                  <i className="manicon manicon-manifold-logo" />
                </figure>
                <button
                  onClick={this.props.hideSignInUpOverlay}
                  className="overlay-close"
                  data-id="overlay-close"
                >
                  Cancel
                  <i className="manicon manicon-x" />
                </button>
              </div>
            </div>
          </header>
          <div className="overlay-content focus">
            <div className="container">
              <div className="inner">
                {this.renderChild()}
              </div>
            </div>
          </div>
        </div>
      </HigherOrder.BodyClass>
    );
  }
}
