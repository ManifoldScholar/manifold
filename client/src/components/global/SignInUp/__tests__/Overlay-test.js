import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import Overlay from '../Overlay';
import build from 'test/fixtures/build';
import { Provider } from 'react-redux';

describe("Global.SignInUp.Overlay component", () => {

  const store = build.store();
  const fakeDomEvent = { stopPropagation: () => undefined, preventDefault: () => undefined };

  const hideOverlayMock = jest.fn();
  const user = build.entity.user("1");

  const root = (
    <Provider store={store}>
      <Overlay
        dispatch={store.dispatch}
        visible
        hideSignInUpOverlay={hideOverlayMock}
        authentication={{
          currentUser: user
        }}
      />
    </Provider>
  );

  it('renders correctly', () => {
    const component = renderer.create(root);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should trigger hideSignInUpOverlay callback when close overlay is clicked', () => {
    const wrapper = mount(root);
    hideOverlayMock.mockClear();
    wrapper.find('[data-id="overlay-close"]').first().simulate('click', fakeDomEvent);
    expect(hideOverlayMock).toHaveBeenCalled();
  });
});
