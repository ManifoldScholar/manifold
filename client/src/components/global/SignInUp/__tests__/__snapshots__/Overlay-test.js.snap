// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global.SignInUp.Overlay component renders correctly 1`] = `
<div
  className="overlay-full-primary"
>
  <header
    className="overlay-header"
  >
    <div
      className="container"
    >
      <div
        className="rel"
      >
        <figure
          className="logo"
        >
          <i
            className="manicon manicon-manifold-logo"
          />
        </figure>
        <button
          className="overlay-close"
          data-id="overlay-close"
          onClick={[MockFunction]}
        >
          Cancel
          <i
            className="manicon manicon-x"
          />
        </button>
      </div>
    </div>
  </header>
  <div
    className="overlay-content focus"
  >
    <div
      className="container"
    >
      <div
        className="inner"
      >
        <div>
          <form
            method="post"
            onSubmit={[Function]}
          >
            <div
              className="row-1-p"
            >
              <div
                className="form-input form-error"
              >
                <label>
                  Email
                </label>
                <input
                  id="login-email"
                  onChange={[Function]}
                  placeholder="Email"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="row-1-p"
            >
              <div
                className="form-input"
              >
                <label>
                  Password
                </label>
                <input
                  id="login-password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
            </div>
            <div
              className="row-1-p"
            >
              <div
                className="form-input"
              >
                <input
                  className="button-secondary button-with-room"
                  type="submit"
                  value="Log in"
                />
              </div>
            </div>
          </form>
          <p
            className="login-links"
          >
            <button
              data-id="show-forgot"
              onClick={[Function]}
            >
              Forgot your password?
            </button>
            <button
              data-id="show-create"
              onClick={[Function]}
            >
              Need to sign up?
            </button>
          </p>
          <section
            className="login-external"
          >
            <div />
            <button
              className="button-secondary-dark"
              onClick={[Function]}
            >
              <i
                className="manicon manicon-envelope"
              />
              <span>
                Log in with Google
              </span>
            </button>
            <button
              className="button-secondary-dark"
              onClick={[Function]}
            >
              <i
                className="manicon manicon-twitter"
              />
              <span>
                Log in with Twitter
              </span>
            </button>
          </section>
        </div>
      </div>
    </div>
  </div>
</div>
`;
